DNS leakage

I wanted to use your application as Socks5 Proxy on your server.
Carefully packed in Docker container.
Detected such effects:
- DNS requests are sent not through the WG tunnel (DNS leakage)
- ~~There are no options to specify 0.0.0.0 as the SOCKS5 server address, thereby making the server on all network cards~~

I rechecked.
In case of such a parameter 
```
[Socks5]
BindAddress = 0.0.0.0:25344
```
i got an error:
```
panic: listen tcp 0.0.0.0:25344: bind: address already in use
goroutine 52 [running]:
log.Panic({0x40001affb8, 0x1, 0x1})
	/usr/local/go/src/log/log.go:354 +0x98
main.socks5Routine.func1(0x400006e140)
	/go/wireproxy/main.go:216 +0x15c
created by main.main
	/go/wireproxy/main.go:391 +0x234
```

Apparently this is a feature of docker.
In the case below, everything is fine. 
```
[Socks5]
BindAddress = 0.0.0.0:2534
```

P.S. Can I PR with `Dockerfile` to your repository? When I'm done with it.

--- a/main.go
+++ b/main.go
@@ -11,6 +11,8 @@ import (
     "os"
     "strings"
     "strconv"
+    "context"
+    "math/rand"
 
     "github.com/armon/go-socks5"
 
@@ -31,7 +33,31 @@ type DeviceSetting struct {
     deviceAddr  *netip.Addr
 }
 
+type NetstackDNSResolver struct {
+    tnet *netstack.Net
+}
+
 type Configuration []ConfigSection
+
+func (d NetstackDNSResolver) Resolve(ctx context.Context, name string) (context.Context, net.IP, error) {
+    addrs, err := d.tnet.LookupContextHost(ctx, name)
+    if err != nil {
+        return ctx, nil, err
+    }
+
+    size := len(addrs)
+    if size == 0 {
+        return ctx, nil, errors.New("no address found for: " + name)
+    }
+
+    addr := addrs[rand.Intn(size)]
+    ip := net.ParseIP(addr)
+    if ip == nil {
+        return ctx, nil, errors.New("invalid address: " + addr)
+    }
+
+    return ctx, ip, err
+}
 
 func configRoot(config Configuration) map[string]string {
     for _, section := range config {
@@ -206,7 +232,7 @@ func socks5Routine(config map[string]string) (func(*netstack.Net), error) {
     }
 
     routine := func(tnet *netstack.Net) {
-        conf := &socks5.Config{ Dial: tnet.DialContext }
+        conf := &socks5.Config{ Dial: tnet.DialContext, Resolver: NetstackDNSResolver{ tnet: tnet } }
         server, err := socks5.New(conf)
         if err != nil {
           log.Panic(err)

